docs/dev/core/internal/payload.md

```diff
@@ -0,0 +1,39 @@
+# Payload format
+
+The main sections of a Glean ping are described in [Ping Sections](../../user/pings/index.md#Ping-sections).
+This **Payload format** chapter describes details of the ping payload that are less relevant to end users of Glean.
+
+## JSON Schema
+
+Glean's ping payloads have a formal JSON schema defined in the [mozilla-pipeline-schemas](https://github.com/mozilla-services/mozilla-pipeline-schemas/) project.
+It is written as a set of [templates](https://github.com/mozilla-services/mozilla-pipeline-schemas/tree/master/templates/include/glean) that are expanded by the mozilla-pipeline-schemas build infrastructure into a [fully expanded schema](https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/master/schemas/glean/baseline/baseline.1.schema.json).
+
+## Metric types
+
+TODO: Fill in the rest of the metric types.
+
+### Event
+
+Events are represented as an array of objects, with one object for each event.
+Each event object has the following keys:
+
+- `timestamp`: (integer) A monotonically increasing timestamp value, in milliseconds.  
+  The first timestamp in the array is always zero, and subsequent timestamps in the array are relative to that reference point.
```

Comment:
It is zero? Isn't it the time since launch? Or an adjusted timestamp?
(See, if only we had documentation ... wait, this _is_ the documentation!)

Reply:
It's definitely zero.  The API we use from Android is time since boot, but for privacy reasons we don't want to leak that information at that resolution.

Reply:
This *is* different from pre-Glean telemetry, though.  Adding a short comment about rationale here.

Reply:
Oh, I see. TIL.	(I checked your event implementation again to understand it)